### ISSUE TITLE
Best performance weights

### ISSUE THREAD

**Issue by robeendra** [haozh7109/SEGAN-TensorFlow2#1]
This code saves the generator and discriminator weights on every epoch during training of the model. Is the weight of last epoch is best? How can we check the performance of the trained model? What should be the loss value of generator and discriminator for best performance ?

**Comment by haozh7109** [comment]
Hi, the model obtained from the last epoch does not necessarily mean the best model, considering that the model may be overfitting. 

According to my actual experience in GAN training, you can see the convergence of the generator and discriminator through the loss, but more importantly, you need to evaluate the performance of the model using inference on the test data, where you can calculate various metrics for the selection of the optimal model.

**Comment by robeendra** [comment]
Hello sir,
Thank you very much for your response. I am new in machine learning and
still learning about GAN from basic. Now I am clear about the evaluation of
model. And I have one more query. Is the performance more better using
LeakyReLU activation instead of PReLU? The SEGAN original paper uses PReLU
in generator part.

Thank you.

**Comment by haozh7109** [comment]
I haven't specifically compared Prelu and leaky relu in SEGAN. But maybe parameterized relu gives better results, or introduces instability because it comes from training. You can run enough experiments to compare the results.

**Comment by robeendra** [comment]
Thank you for your suggestion. Can you suggest any evaluation tool? ITU
said PESQ is also outdated. POLQA is licensed. Is there any other option?
